service/src/main/java/bio/terra/workspace/db/WorkspaceDao.java

```diff
@@ -151,44 +154,38 @@ public Workspace getWorkspace(UUID id) {
   }
 
   @WriteTransaction
-  public boolean updateWorkspace(UUID workspaceId, String name, String description) {
-    if (name == null && description == null) {
-      throw new MissingRequiredFieldException("Must specify name or description to update.");
+  public boolean updateWorkspace(
+      UUID workspaceId, String name, String description, Map<String, String> propertyMap) {
```

Comment:
could there be a peropertyMap is not null but empty situation?

Reply:
e,g, some fields are invalid and get filtered

Reply:
also annotate with nullable

Reply:
It is not invalid to have an empty property map. If you wanted to remove all of the properties, that is what you would provide.